src/wallet/rpcwallet.cpp

```diff
@@ -477,12 +496,10 @@ UniValue listaddresses(const UniValue& params, bool fHelp)
 
                 UniValue sapling_obj(UniValue::VOBJ);
 
-                if (source == LegacyHDSeed) {
-                    std::string hdKeypath = pwalletMain->mapSaplingZKeyMetadata[ivk].hdKeypath;
-                    std::optional<unsigned long> accountId = libzcash::ParseZip32KeypathAccount(hdKeypath);
-
-                    if (accountId.has_value()) {
-                        sapling_obj.pushKV("zip32AccountId", (uint64_t) accountId.value());
```

Comment:
Hmm, this is a breaking change to the RPC. At a minimum, we should continue to add this field if `source == LegacyHDSeed`.

Reply:
It's a breaking change to a just-released hotfix RPC; if we'd been thinking properly we'd have just used the keypath in either case. I think that breaking compatibility in this sort of situation is fine, it's highly unlikely that the account ID is being used in any mission-critical fashion right now.

Reply:
Let's document this in the release notes.